table/types/types.go

```diff
@@ -247,3 +247,13 @@ func ConvertType(type_ ColumnType) (gpb.ColumnDataType, error) {
 	}
 
 }
+
+// WriteOp is the type of write operation
+// current supported [Insert, Delete]
+// TODO: [Update]
+type WriteOp uint
```

Comment:
Is it ok to get rid of the WriteOp concept, and directly define Write and Delete functions? Which means, the `Build(WriteOp)` method of `Request`, has to be renamed to `ToDeleteRequest`, or `ToInsertRequest` or other proper names.

Reply:
Thanks @yuanbohan review and advice.I still need some help for figure out it.
 
do you mean change the `Request` to this?
```golang
interface Request{
    ...
}

type ToDeleteRequest struct {
   ...
}

type ToInsertRequest struct {
   ...
}

```

But if do this, we need implement twice the functions for `ToDeleteRequest` and `ToInsertRequest`.  Cloud you told me the benefits in this implements?

Or we can add `WriteOp` parameter in `Request`, do you think  is good method?
```golang

type Request struct {
	header *header.Header
	tables []*table.Table
        writeOp types.WriteOp
}

func write(...,writeOp,...) ... {
	...
	request_, err := request.New(header_, writeOp,tables...).Build()
        ...
}
```

Thanks for your time to help me improve code.

Reply:
I think the latter one is a better option :+1:

BTW, WriteOp is somehow confusing for delete is not a write operation, maybe direct `Operation`? or some other good names.

Reply:
Done, please re-check the code, thank you.